libsawtooth/src/client/rest.rs

```diff
@@ -149,6 +152,116 @@ impl SawtoothClient for RestApiSawtoothClient {
+    ) -> Result<(), SawtoothClientError> {
+        let url = format!("{}/batches", &self.url);
+
+        let mut batch_file = File::open(filename).map_err(|err| {
+            SawtoothClientError::new_with_source("failed to open batch file", err.into())
+        })?;
+
+        let batch_list: BatchList =
+            protobuf::parse_from_reader(&mut batch_file).map_err(|err| {
+                SawtoothClientError::new_with_source("unable to parse file contents", err.into())
+            })?;
+        let len = batch_list.batches.len();
+        let batch_ids = batch_list
+            .batches
+            .iter()
+            .map(|batch| batch.header_signature.clone())
+            .collect();
+
+        let batch_lists = split_batches(batch_list, size_limit);
+
+        let start = Instant::now();
+
+        for list in batch_lists.into_iter() {
+            let submit_list = list.write_to_bytes().map_err(|err| {
+                SawtoothClientError::new_with_source(
+                    "failed to get bytes from batch list",
+                    err.into(),
+                )
+            })?;
+            let request = Client::new()
+                .post(&url)
+                .header(header::CONTENT_TYPE, "application/octet-stream")
+                .body(submit_list);
+            request.send().map_err(|err| {
+                SawtoothClientError::new_with_source("request failed", err.into())
+            })?;
+        }
+        println!(
+            "batches: {},  batch/sec: {}",
+            len,
+            len as f64 / (((start.elapsed()).as_millis() as f64) / 1000f64)
+        );
+
+        if let Some(wait) = wait {
+            wait_for_batches(&self.url, batch_ids, wait)
+        } else {
+            Ok(())
+        }
+    }
+}
+
+/// Check the status of the batches with the given ids, wait the requested amount of time
+/// for the batches to commit
+fn wait_for_batches(
+    url: &str,
+    ids: Vec<String>,
+    wait: Duration,
+) -> Result<(), SawtoothClientError> {
+    let end_time = Instant::now()
+        .checked_add(wait)
+        .ok_or_else(|| SawtoothClientError::new("failed to schedule timeout"))?;
+    let start_time = Instant::now();
+    loop {
+        let url = format!(
+            "{}/batch_statuses?id={}&wait={:?}",
+            url,
+            ids.join(","),
+            wait.as_secs()
+        );
+        let error_msg = "unable to get statuses";
+        let statuses = get::<StatusList>(&url, error_msg)?;
+
+        if let Some(status_list) = statuses {
+            let any_pending_batches = status_list
+                .data
+                .iter()
+                .any(|info| &info.status == "PENDING");
+            if any_pending_batches {
+                if Instant::now() < end_time {
+                    continue;
+                } else {
+                    println!("wait timed out! Some batches have not yet been committed...\nid  {}\ninvalid_transactions  {:?}\nstatus  {}",
```

Comment:
These need to be return values from the library - it's up to the caller to decide how to log/print these.  It would probably make sense to return an enum:

```
enum BatchWaitResult {
    BatchesCommitted,
    BatchesRemaing(Vec<String>)
}
```

Not sure how this should be propagated up the stack, though.

Reply:
It's also more likely that this isn't the concern of the library, as a library consumer could create this behaviour themselves.

Reply:
Fixed, `submit_batches` no longer calls `wait_for_batches`